quickwit-doc-mapping/src/default_doc_mapper/property_entry.rs

```diff
@@ -0,0 +1,413 @@
+            }
+            _ => bail!("Expected object or array"),
+        };
+        Ok(parsed_values)
+    }
+
+    pub fn parse_u64(
+        &self,
+        json_value: &JsonValue,
+        options: &IntOptions,
+    ) -> anyhow::Result<Vec<(FieldPath, Value)>> {
+        let parsed_values = match json_value {
+            JsonValue::Array(array) => {
+                // TODO: check if cardinality is single or not
+                // TODO: the chain seems too long, is it possible to make it simpler?
+                array
+                    .iter()
+                    .map(|element| self.parse_u64(&element, options))
+                    .collect::<Result<Vec<_>, _>>()?
+                    .into_iter()
+                    .flatten()
+                    .collect()
+            }
+            JsonValue::Number(value_as_number) => {
+                if let Some(value_as_u64) = value_as_number.as_u64() {
+                    vec![(
+                        FieldPath::new(&self.name),
+                        Value::U64(value_as_u64.to_owned()),
+                    )]
+                } else {
+                    bail!("Expected u64")
+                }
+            }
+            _ => bail!("Expected array or number"),
+        };
+        Ok(parsed_values)
+    }
+
+    pub fn parse_text(
+        &self,
+        json_value: &JsonValue,
+        options: &TextOptions,
+    ) -> anyhow::Result<Vec<(FieldPath, Value)>> {
+        let parsed_values = match json_value {
+            JsonValue::Array(array) => {
+                // TODO: check if cardinality is single or not
+                array
+                    .iter()
+                    .map(|element| self.parse_text(&element, options))
+                    .collect::<Result<Vec<_>, _>>()?
+                    .into_iter()
+                    .flatten()
+                    .collect()
+            }
+            JsonValue::String(value_as_str) => {
+                vec![(
+                    FieldPath::new(&self.name),
+                    Value::Str(value_as_str.to_owned()),
+                )]
+            }
+            _ => bail!("Expected object or array"),
+        };
+        Ok(parsed_values)
+    }
+}
+
+pub struct FieldPath<'a> {
+    components: Vec<&'a str>,
+}
+
+impl<'a> FieldPath<'a> {
+    pub fn new(path: &'a str) -> Self {
+        Self {
+            components: vec![path],
+        }
+    }
+
+    pub fn with_parent(mut self, parent: &'a str) -> Self {
+        self.components.insert(0, parent);
+        self
+    }
+
+    pub fn field_name(&self) -> String {
+        self.components.join("__dot__")
+    }
+}
+
+// Struct used for serialization and deserialization
+// Main advantage: having a flat structure and gain flexibility
+// if we want to add some syntaxic sugar in the mapping.
+// Main drawback: we have a bunch of mixed parameters in it but
+// seems to be reasonable.
+#[derive(Clone, Serialize, Deserialize)]
+struct PropertyEntryParams {
```

Comment:
Should we be even more explicit, and rename it PropertyEntryForSerialization

Reply:
done